### ISSUE TITLE
FC Critical bugs

### ISSUE THREAD

**Issue by Martijn1981** [modxbot/migrate#4207]
Martijn1981 created Redmine issue ID 4207

FC still has a few major bugs. I want to customize a template for the client, so that the page is automatically hidden from all menu's and automatically published.

However
- the page refuses to be automatically published
- the page is not hidden from the menu's.

See my screenshot. I've put a "1" in their default values, but changing the default value doesn't seem to work. The only solution for me now is to make these fields visible, so that the client has to manually hide the page from the menu's and publish it as well. Of course I've used resource-create AND resource-update for this template.

Also, when I'm using Parent Resource to appoint another parent: the page reloads, but the Parent is NOT saved inside the Parent Resource field. It's empty!

**Comment by Martijn1981** [comment]
Martijn1981 submitted:

Also, when you create a new document, fill in a menutitle and when you leave that page by clicking on another page, the new document is automatically saved.

**Comment by Martijn1981** [comment]
Martijn1981 submitted:

One last thing regarding the screenshot: it doesn't matter whether I put a 0, 1 or nothing there: the template isn't automatically published and it's not hidden from the menu. All these bugs only seem to popup whenever I'm creating/modifying new FC rules for a different, lower, user-group. 

I always make a user-group called Webmasters for the client, duplicate the Content Editor policy and edit it where it's needed and I'm always linking 1 FC Profile to that user-group, which contains resource-create and resource-update customizations for every template the client has access too. 

The FC parts (in modx-resource-settings) that still seem buggy to me are:
- template (sometimes the TinyMCE doesn't load in the standard template when I'm using FF4.)
- published (it seems kinda random. It works sometimes. Not all the time)
- parent-cmb (when I click on Parent Resource, choose another node in the tree: the page does reload properly, but there's nothing to be seen in this field afterwards)
- hidemenu (I want to hide a few templates from menus, but it seems I can't properly enable this function all the time)

Also: the tab Template Variables (in Set: resource/update as well as Set: resource/create) seems to be faulty. I've unchecked a HTML area TV a couple of times now, but it still seems to popup in templates now and then. The only way to get rid of it, is to completely empty the tab-field (containing modx-resource-settings, for example) it seems. 

You guys did a great job on the other parts of 2.1. Seriously. However, it seems that Form Customization has gone a bit downhill. But this is still RC and I hope to contribute with this detailed post.

**Comment by splittingred** [comment]
splittingred submitted:

Your issue seems to be due to the checkbox fixes we applied in 2.1. I've fixed this issue:

https://github.com/modxcms/revolution/commit/6f610fc1b1ec386b983d5ae6cc40a69e62c1335d

It's worth it to note, that until we convert Form Customization (specifically the loading of the rules and forms) into PHP, rather than using JS hooks, it might be a trifle bit unsteady. There's an issue for this here: #3113 It's definitely not trivial, however, to do that.

We'll have this specific issue fixed for 2.1-rc2/pl, though.
